test/IntegrationTests/test_deletechat.py

```diff
@@ -0,0 +1,299 @@
+        self.assertEqual(response_data["message"], "Chat deleted successfully", "Unexpected message in response.")
+
+        logger.info("=== Test Case 2 Completed ===\n")
+
+    def test_customer_guid_value_missing_and_correct_chat_id(self):
+        """Test case 3: customer_guid value missing and correct chat_id"""
+        logger.info("=== Starting Test Case 3: customer_guid value missing and correct chat_id ===")
+
+        url = f"{self.BASE_URL}/deletechat"
+        data = {
+            "chat_id": self.valid_chat_id
+        }
+        logger.info(f"INPUT : Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        self.assertEqual(response.status_code, 422)
+        self.assertIn("field required", response.text.lower())
+        logger.info("=== Test Case 3 Completed ===\n")
+
+    def test_correct_customer_guid_and_missing_chat_id(self):
+        """Test case 4: correct customer_guid and missing chat_id"""
+        logger.info("=== Starting Test Case 4: correct customer_guid and missing chat_id ===")
+
+        url = f"{self.BASE_URL}/deletechat"
+        data = {
+            "chat_id": self.valid_chat_id
+        }
+        logger.info(f"INPUT : Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        self.assertEqual(response.status_code, 422)
+        self.assertIn("field required", response.text.lower())
+        logger.info("=== Test Case 4 Completed ===\n")
+
+    def test_empty_customer_guid_and_empty_chat_id(self):
+        """Test case 5: Empty customer_guid and empty chat_id"""
+        logger.info("=== Starting Test Case 5: Empty customer_guid and empty chat_id ===")
+
+        url = f"{self.BASE_URL}/deletechat"
+        data = {}
+        logger.info(f"INPUT: Sending request with data:\n{str(data)}")
+
+        response = requests.post(url, json=data)
+        logger.info(f"OUTPUT: Response status code: {response.status_code}")
+        logger.info(f"OUTPUT: Response content: {response.text}")
+
+        self.assertEqual(response.status_code, 422)  # Expecting 422 for missing fields
+        self.assertIn("field required", response.text.lower())  # Check that the response indicates required fields are missing
+
+        logger.info("=== Test Case 5 Completed ===\n")
+
+    def test_duplicate_deletion(self):
+        """Test case 6: Duplicate Deletion with valid customer_guid and valid chat_id"""
+        logger.info("=== Starting Test Case 6: Duplicate Deletion ===")
+
+        url = f"{self.BASE_URL}/deletechat"
+        data = {
+            "customer_guid": self.valid_customer_guid,
+            "chat_id": self.valid_chat_id
```

Comment:
You are reusing the class chat_id

You need to first create a chat id and then get the chat_id